### PR TITLE
updated signUnsignedJWT to sign self issued JWT

### PR DESCRIPTION
**Pull request by vijayshiyani**

### CODE REVIEW

File: packages/sdk/src/util/jwt.ts [hunk 1]
```diff
@@ -1,14 +1,30 @@
...
+  if (decoded.payload.iss === SELF_ISSUED_V2 || decoded.payload.iss === SELF_ISSUED_V2_VC_INTEROP) {
+    if (!signer) throw new Error('missing_signer: No Signer functionality has been configured');
+    if (!decoded.header.typ) decoded.header.typ = 'JWT';
+    if (!decoded.header.alg) decoded.header.alg = 'EdDSA';
+    const timestamps: Partial<JWTPayload> = {
+      iat: Math.floor(Date.now() / 1000),
+    };
+
+    const fullPayload = { ...timestamps, ...decoded.payload };
+    return createJWS(fullPayload, signer, decoded.header);
```
Comment (linasi) [inline packages/sdk/src/util/jwt.ts:26]: @vijayshiyani why do we need to call `createJWS` method in case of different `iss` attribute value?

Comment (linasi) [reply packages/sdk/src/util/jwt.ts:26]: Wouldn't explicitly specifying issuer at:
```
return createJWT(decoded.payload, { issuer: SELF_ISSUED_V2, signer }, decoded.header);
```
do the same job?

**Comment by linasi** [comment]
@vijayshiyani could you also bump up the version of the SDK package?

**Comment by linasi** [comment]
@vijayshiyani also please add a test or two for this new logic

**Comment by vijayshiyani** [comment]
> 

done @linasi

**Review by linasi — APPROVED** [review]
@vijayshiyani LGTM.
Please take a look at a few other not so urgent comments I have left.

File: packages/sdk/src/util/jwt.ts (hunk 1 above)
Comment (vijayshiyani) [reply packages/sdk/src/util/jwt.ts:26]: 🤯  thanks for picking this up. why wouldn't I think about this before? 
yes it works. I will reject this PR. we dont need this. thanks. 👍 
@linasi

Comment (linasi) [reply packages/sdk/src/util/jwt.ts:26]: @vijayshiyani how did you solve it exactly? I see new PR does not contain any changes to this logic.

Comment (vijayshiyani) [reply packages/sdk/src/util/jwt.ts:26]: SDK users can call `signUnsignedJWT` function with `SELF_ISSUED_V2_VC_INTEROP` as issuer DID. function will just sign JWT and keep iss attribute to  SELF_ISSUED_V2_VC_INTEROP 

e.g.
`  const SELF_ISSUED_V2_VC_INTEROP = 'https://self-issued.me/v2/openid-vc';
      const signedIdToken = await signUnsignedJWT(
        tokenResponse.unsigned_id_token,
        SELF_ISSUED_V2_VC_INTEROP,
        new Ed25519(privateKey.toBytes())
      );`

Comment (linasi) [reply packages/sdk/src/util/jwt.ts:26]: Perfect, thanks!

**Pull request closed without merging**